source/Engine.cpp

```diff
@@ -131,11 +152,36 @@ namespace {
 		}
 	}
 
+	template <class Type>
+	void Prune(list<Type> &objects)
+	{
+		for(auto it = objects.begin(); it != objects.end(); )
+		{
+			if(it->ShouldBeRemoved())
+				it = objects.erase(it);
+			else
+				++it;
+		}
+	}
+
 	template <class Type>
 	void Append(vector<Type> &objects, vector<Type> &added)
 	{
-		objects.insert(objects.end(), make_move_iterator(added.begin()), make_move_iterator(added.end()));
-		added.clear();
+		if(added.empty())
+			return;
+		if(objects.empty())
+			objects.swap(added);
+		else if(objects.size() <= added.size())
+		{
+			objects.swap(added);
+			objects.insert(objects.end(), make_move_iterator(added.begin()), make_move_iterator(added.end()));
+			added.clear();
+		}
+		else
+		{
+			objects.insert(objects.end(), make_move_iterator(added.begin()), make_move_iterator(added.end()));
+			added.clear();
+		}
```

Comment:
this is an unrelated performance upgrade isnt it?

Reply:
Not entirely unrelated, but it's not essential to the PR.

Reply:
I mean it is your decision, but stuff is faster to review if it is split up. but then again, it is just a small change